### PR TITLE
generate the API documentation

### PR DESCRIPTION
**Pull request by fdiblen**
**Description**

Sets up API documentation which will be automatically built by readthedocs.
Describes how to build the documentation locally.

**Related issues**:

- #28 

**Instructions to review the pull request**

- Visit https://cudawrappers.readthedocs.io/en/28-api-docs/ to see the generated documentation.

### CODE REVIEW

File: README.dev.md [hunk 1]
```diff
@@ -192,9 +192,38 @@ where `<tool>` can be any of the following:
 
 :construction:
 
-## Building the documentation
+## Building the API documentation
 
-:construction: See issue #29
+The API documentation is automatically generated for `main` branch and the pull requests to be merged to `main` branch.
+The documentation is hosted at <https://cudawrappers.readthedocs.io/en/latest/> and is automatically built by readthe docs service.
```
Comment (abelsiqueira) [inline README.dev.md:198]:
```suggestion
The documentation is hosted at [https://cudawrappers.readthedocs.io/en/latest/](https://cudawrappers.readthedocs.io/en/latest/) and is automatically built by read the docs service.
```

File: README.dev.md [hunk 2]
```diff
@@ -192,9 +192,38 @@ where `<tool>` can be any of the following:
...
+
+### Building locally 
+
+To build the documentation locally, you will need to follow dependencies.
```
Comment (abelsiqueira) [inline README.dev.md:202]:
```suggestion
To build the documentation locally, you will need the following dependencies. 
```

File: README.dev.md [hunk 3]
```diff
@@ -192,9 +192,38 @@ where `<tool>` can be any of the following:
...
 
-:construction: See issue #29
+The API documentation is automatically generated for `main` branch and the pull requests to be merged to `main` branch.
+The documentation is hosted at <https://cudawrappers.readthedocs.io/en/latest/> and is automatically built by readthe docs service.
+
+### Building locally 
+
+To build the documentation locally, you will need to follow dependencies. 
+
+- doxygen
```
Comment (abelsiqueira) [inline README.dev.md:204]:
```suggestion
- [doxygen](https://www.doxygen.nl/index.html)
```

File: README.dev.md [hunk 4]
```diff
@@ -192,9 +192,38 @@ where `<tool>` can be any of the following:
...
+- Python packages:
+  - sphinx
+  - breathe
+  - exhale
+  - myst-parser
```
Comment (abelsiqueira) [inline README.dev.md:209]: Can you add links to the others as well?

File: README.dev.md [hunk 5]
```diff
@@ -192,9 +192,38 @@ where `<tool>` can be any of the following:
...
+
+The Python dependencies can be found in `docs/requirements.txt`.
+
+To build the documentation run:
+
+```shell
+python3 -m venv venv
+. ./venv/bin/activate
```
Comment (abelsiqueira) [inline README.dev.md:217]:
```suggestion
source ./venv/bin/activate
```
I think it is better to leave explicit.

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
Thanks for the PR. I made some suggestions.
Also, we should consider looking into hosting on GH-pages to avoid ADs (we can spawn an issue).

File: README.dev.md (hunk 1 above)
Comment (fdiblen) [inline README.dev.md:198]:
```suggestion
The documentation is hosted at <https://cudawrappers.readthedocs.io/en/latest/> and is automatically built by readthedocs service.
```

Comment (abelsiqueira) [reply README.dev.md]: We discussed that `<link>` automatically adds a link, so my first comment can be ignored.
The second comment we'll use the service name without spaces.

File: README.dev.md [hunk 6]
```diff
@@ -192,9 +192,38 @@ where `<tool>` can be any of the following:
...
+### Building locally 
+
+To build the documentation locally, you will need the following dependencies. 
+
+- [doxygen](https://www.doxygen.nl/index.html)
+- Python packages:
+  - sphinx
+  - breathe
+  - exhale
+  - myst-parser
```
Comment (fdiblen) [inline README.dev.md:209]:
```suggestion
  - [sphinx](https://www.sphinx-doc.org)
  - [breathe](https://breathe.readthedocs.io)
  - [exhale](https://exhale.readthedocs.io)
  - [myst-parser](https://myst-parser.readthedocs.io)
```

File: README.dev.md (hunk 5 above)
Comment (abelsiqueira) [reply README.dev.md:217]: They are different: https://unix.stackexchange.com/questions/312573/what-is-the-difference-between-and-source

**Review by abelsiqueira — APPROVED** [review]
Looks great, thanks

**Review by abelsiqueira — APPROVED** [review]

**Pull request merged**